internal/controller/stas/containerimagescan_status.go

```diff
@@ -28,8 +36,80 @@ func newContainerImageStatusPatch(cis *stasv1alpha1.ContainerImageScan) *stasv1a
 		}
 	}
 
-	return stasv1alpha1ac.ContainerImageScan(cis.Name, cis.Namespace).
-		WithStatus(status)
+	return &containerImageScanStatusPatch{
+		cis: cis,
+		patch: stasv1alpha1ac.ContainerImageScan(cis.Name, cis.Namespace).
+			WithStatus(status),
+	}
+}
+
+type containerImageScanStatusPatch struct {
+	cis             *stasv1alpha1.ContainerImageScan
+	patch           *stasv1alpha1ac.ContainerImageScanApplyConfiguration
+	vulnerabilities []stasv1alpha1.Vulnerability
+	minSeverity     *stasv1alpha1.Severity
+}
+
+func (b *containerImageScanStatusPatch) withCondition(c *metav1ac.ConditionApplyConfiguration) *containerImageScanStatusPatch {
+	b.patch.Status.
+		WithConditions(NewConditionsPatch(b.cis.Status.Conditions, c)...)
+	return b
+}
+
+func (b *containerImageScanStatusPatch) fromScanJob(job *batchv1.Job) *containerImageScanStatusPatch {
+	b.patch.Status.
+		WithLastScanTime(metav1.Now()).
+		WithLastScanJobUID(job.UID)
+
+	return b
+}
+
+func (b *containerImageScanStatusPatch) fromCompletedScanJob(job *batchv1.Job, vulnerabilities []stasv1alpha1.Vulnerability, minSeverity stasv1alpha1.Severity) *containerImageScanStatusPatch {
```

Comment:
Not sure about the `fromX` names. Based on the name alone, I would think it creates a new patch from the _from_ thing alone, not updating an existing one. `withCompleteScanJob` or something would make more sense to me.

Reply:
Agreed, and it used to be like that. Updated now.